### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Abe1able**
### In this milestone of the project, I have

-  Created a class named Classroom with respective requirements.
-  Formed a relationship between Classroom and Student classes.
-  Created a class named Book that will be used to hold title and author of a book.
-  Created a class named Rental for holding rentals.
-  Formed a many-to-many relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -9,10 +9,11 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (AshakaE) [inline person.rb]: If we set `rentals` as an `attr_reader`, it means we can only read rentals but not write rentals, ultimately we need rentals to be read and writable

- [x] Kindly update this attribute

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (AshakaE) [inline book.rb]: Also here, we want to make sure we can make a rental, not only read what was rented

- [x] Kindly update attribute for `rentals`

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @Abe1able  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Code base looks neat 👏 
- Looks promising so far 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Abe1able) [reply person.rb]: Understood. I'll implement that. Thank you.

File: book.rb (hunk 2 above)
Comment (Abe1able) [reply book.rb]: Sure thing. Thank you for your detailed review @AshakaE

File: person.rb (hunk 1 above)
Comment (Whoistolu) [reply person.rb]: - Well resolved 👍🏽

File: book.rb (hunk 2 above)
Comment (Whoistolu) [reply book.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Abe1able 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**